Source/Directory.h

```diff
@@ -53,94 +24,148 @@ class Directory
      * description - description
      * subpath - again source dependent but used for finding the module
      */
-
+	
     /**
      * Opens a directory with the given URL, downloading the latest version of
      * the contents. 
-     */
-    Directory (URL location)
-    {
-        DownloadCache cache;
-        String file = cache.downloadTextFile (location);
-        ScopedPointer<XmlElement> xml = XmlDocument (file).getDocumentElement();
+	*/
+    Directory(URL location)
+	{
+		DownloadCache cache;
+		String file = cache.downloadFromDatabase();
 
-        if (xml)
+		if (!file.isEmpty())
         {
-            directory = ValueTree::fromXml (*xml);
+			directory = Database::parseToArray (file);
         }
-        else
+		else
         {
             throw JpmFatalExcepton ("directory format error or network problem",
                                     "Check " 
                                     + cache.getCachedFileLocation (location).getFullPathName() 
                                     + " for debugging which should contain the contents of " 
                                     + location.toString (true));
-        }
+		}
+	}
     }
```

Comment:
Sorry, thought I’d committed those fixes, try again now. 

> On 10 Dec 2015, at 20:07, Jim <notifications@github.com> wrote:
> 
> In Source/Directory.h <https://github.com/jcredland/jpm/pull/1#discussion_r47277812>:
> 
> >          {
> >              throw JpmFatalExcepton ("directory format error or network problem",
> >                                      "Check " 
> >                                      + cache.getCachedFileLocation (location).getFullPathName() 
> >                                      + " for debugging which should contain the contents of " 
> >                                      + location.toString (true));
> > -        }
> > +		}
> > +	}
> >      }
> This might be an error? I think this is the cause of one of the build errors?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/jcredland/jpm/pull/1/files#r47277812>.
>